src/ptracer.rs

```diff
@@ -189,6 +299,32 @@ impl Tracee {
         }
     }
 
+    #[cfg(all(target_os = "android", target_arch = "aarch64"))]
+    pub fn set_debug_register(&self, regtype: DebugRegisterType, index: usize, data: user_hwdebug_state_reg) -> Result<()> {
+        assert!(index < 16);
+        let mut registers = std::mem::MaybeUninit::uninit();
+        let mut rv = regsvec {
+            ufb: &mut registers as *mut _ as *mut libc::c_void,
+            len: std::mem::size_of::<user_hwdebug_state>(),
+        };
+        let res = unsafe {
+            libc::ptrace(PTRACE_GETREGSET, self.pid, regtype, &mut rv as *mut _ as *mut libc::c_void)
+        };
+
+        nix::errno::Errno::result(res)?;
+
+        let mut state: user_hwdebug_state = unsafe { registers.assume_init() };
```

Comment:
It'd be nice to share the unsafe code that reads the `user_hwdebug_state`, even just in a private method.

Reply:
There is no longer a `GETREGSET` in the `set_debug_registers` function.